astropy/io/fits/_utils.pyx

```diff
@@ -0,0 +1,57 @@
+from collections import OrderedDict
+
+cdef Py_ssize_t BLOCK_SIZE = 2880  # the FITS block size
+cdef Py_ssize_t CARD_LENGTH = 80
+cdef str VALUE_INDICATOR = '= '  # The standard FITS value indicator
+cdef str END_CARD = 'END' + ' ' * 77
+
+
+def parse_header(fileobj):
+    """The main method allowing to parse quickly a FITS header."""
+
+    cards = OrderedDict()
+    read_blocks = []
```

Comment:
Lists can also be `cdef`'d. It won't be much faster but the `.append` call is optimized then.

Reply:
Good to know, thanks.